capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,229 @@ async function submitInput() {
+      const response = await callPerspective(substrings[i], lang, requestedAttributes);
+      if (typeof(response.error) != 'undefined') {
+        analysisContainer.removeChild(loadingEl);
+        analysisContainer.appendChild(createAnyElement('p', 'Perspective API was not able to get scores for detailed analysis'));
+        return;
+      }
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
+      const attributes = sortAttributes(response);
+
+      // Set up and color the substring appropriately	
+      const substring = substrings[i];
+      const substringEl = createAnyElement('span', substring);
+      substringEl.onclick = function() { setUpSuggestions(substring, lang); };
+      colorSubstring(substringEl, toxicityScore);
+     
+      // Attach a tooltip (info-box for the substring)
+      const tooltipEl = createTooltip(attributes);
+      substringEl.appendChild(tooltipEl);
+      result.appendChild(substringEl);
+    }
+  }
+  analysisContainer.removeChild(loadingEl);
+  analysisContainer.appendChild(result);
+}
+
+/** Sets up the suggestions module */
+function setUpSuggestions(text, lang) { 
+  // Set up the header containing the selected segment
+  const suggestionsContainer = document.getElementById('suggestions-input-container');
+  suggestionsContainer.innerHTML = '';
+  suggestionsContainer.appendChild(createAnyElement('p', text));
+
+  // Set up the input box for the substring the user wants suggestions for
+  const inputBox = document.createElement('input');
+  inputBox.className = 'form-control';
+  inputBox.setAttribute('id', 'input-box')
+  inputBox.setAttribute('placeholder', 'Input substring of this segment you woud like suggestions for');
+  suggestionsContainer.appendChild(inputBox);
+
+  // Set up the submit button
+  const submitButton = document.createElement('button');
+  submitButton.innerHTML = 'Submit';
+  submitButton.className = 'btn btn-primary';
+  submitButton.onclick = function() { getSuggestions(text, lang); };
+  suggestionsContainer.appendChild(submitButton);
+}
+
+/** Gets the suggestions & their score changes for a subtring based on the Datamuse API */
+async function getSuggestions(text, lang) {
+  // Handle the input from the input box
+  const inputBox = document.getElementById('input-box');
+  if (!inputBox) {
+    return;
+  }
+  const substringForSuggestions = inputBox.value;
+  if (text.indexOf(substringForSuggestions) == -1) {
+  	alert("Substring must be in the selected segment");
+  	return;
+  } else if (substringForSuggestions == '') {
+    alert('Input cannot be empty');
+    return;
+  }
+  inputBox.innerHTML = '';
+  
+  const resultsContainer = document.getElementById('perspective-datamuse-analysis');
+  resultsContainer.innerHTML = '';
+  const perspectiveCallForOriginal = await callPerspective(text, lang, ['TOXICITY']);
+  const toxicityScoreOriginal = perspectiveCallForOriginal.attributeScores.TOXICITY.summaryScore.value;
```

Comment:
We might have to consider error checking this, just in case Perspective returns an error and we get a `NullPointerException`.

Reply:
Fixed in #25